src/core/index.ts

```diff
@@ -63,27 +71,106 @@ const approveChange = async (context: Context): Promise<void> => {
  * @param context The request context created by Probot core.
  * @param startTime The timestamp that this run started.
  */
-const createPassingStatus = async (
+export const createPassingStatus = async (
   context: Context,
   startTime: string,
 ): Promise<void> => {
+  await createStatus(
+    context,
+    "success",
+    "completed",
+    composeStatusCheckTitle(),
+    composeStatusCheckSummary(),
+    composeStatusCheckDetails(),
+    startTime,
+  );
+};
+
+/**
+ * Create a status in pull requets indicating that the run
+ * has started.
+ *
+ * @param context The request context from Probot core.
+ * @param startTime The string time that the run started.
+ */
+export const createStartStatus = async (
+  context: Context,
+  startTime: string,
+): Promise<void> => {
+  await createStatus(
+    context,
+    undefined,
+    "in_progress",
+    composeStatusCheckTitle(),
+    composeStatusCheckSummary(),
+    composeStatusCheckDetails(),
+    startTime,
+  );
+};
+
+export const createCrashStatus = async (
+  context: Context,
+  startTime: string,
+  error: Error,
+): Promise<void> => {
+  await createStatus(
+    context,
+    "failure",
+    "completed",
+    composeCrashReportTitle(),
+    composeCrashReportSummary(),
+    composeCrashReportDetails(error),
+    startTime,
+  );
+};
+
+/**
+ * Creates a check in the pull request.
+ *
+ * @param context The request context from Probot core.
+ * @param conclusion The conclusion of the run.
+ * @param status The status of the run.
+ * @param title The title that shows up when user clicks on the check.
+ * @param summary The summary that shows up under the title.
+ * @param details The details that shows up under the summary.
+ * @param startTime The string time that the run started.
+ */
+export const createStatus = async (
+  context: Context,
+  conclusion:
+    | "success"
+    | "failure"
+    | "neutral"
+    | "cancelled"
+    | "timed_out"
+    | "action_required"
+    | undefined,
+  status: "in_progress" | "completed" | "queued",
+  title: string,
+  summary: string,
+  details: string,
+  startTime: string,
+): Promise<void> => {
+  const completedAt: string | undefined = conclusion
+    ? new Date().toISOString()
+    : undefined;
```

Comment:
Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/manual/tianhaoz/approveman/pullRequest?prid=6092518)